### PR TITLE
Feature/room members perf

### PR DESCRIPTION
**Pull request by ganfra**
This PR tends to get better performance when handling room members and users
This is related to  #375

### CODE REVIEW

File: matrix-sdk-android/src/main/java/im/vector/matrix/android/internal/session/user/UpdateUserTask.kt [hunk 1]
```diff
@@ -38,21 +39,6 @@ internal interface UpdateUserTask : Task<UpdateUserTask.Params, Unit> {
...
-                val roomMember = RoomMembers(realm, roomId).get(userId) ?: return@forEach
-                if (roomMember.membership != Membership.JOIN) return@forEach
-
-                val userEntity = UserEntity.where(realm, userId).findFirst()
-                                 ?: realm.createObject(UserEntity::class.java, userId)
-                userEntity.displayName = roomMember.displayName ?: ""
-                userEntity.avatarUrl = roomMember.avatarUrl ?: ""
-            }
-        }
+        return Try.just(Unit)
```
Comment (bmarty) [inline matrix-sdk-android/src/main/java/im/vector/matrix/android/internal/session/user/UpdateUserTask.kt]: Remove this useless task?

File: matrix-sdk-android/src/main/java/im/vector/matrix/android/internal/database/helper/RoomEntityHelper.kt [hunk 2]
```diff
@@ -37,25 +37,22 @@ internal fun RoomEntity.addOrUpdate(chunkEntity: ChunkEntity) {
...
+    if (stateEvent.eventId == null || (filterDuplicates && fastContains(stateEvent.eventId))) {
+        return
+    } else {
+        val entity = stateEvent.toEntity(roomId).apply {
             this.stateIndex = stateIndex
             this.isUnlinked = isUnlinked
             this.sendState = SendState.SYNCED
         }
-        untimelinedStateEvents.add(0, eventEntity)
+        untimelinedStateEvents.add(entity)
```
[the remark this answers is not in the document]
Comment (ganfra) [reply matrix-sdk-android/src/main/java/im/vector/matrix/android/internal/database/helper/RoomEntityHelper.kt]: Yes, we are not using order

File: matrix-sdk-android/src/main/java/im/vector/matrix/android/internal/session/user/UpdateUserTask.kt (hunk 1 above)
Comment (ganfra) [reply matrix-sdk-android/src/main/java/im/vector/matrix/android/internal/session/user/UpdateUserTask.kt]: Thanks

File: matrix-sdk-android/src/main/java/im/vector/matrix/android/api/session/room/timeline/TimelineEvent.kt [hunk 3]
```diff
@@ -33,7 +33,7 @@ data class TimelineEvent(
         val localId: Long,
         val displayIndex: Int,
         val senderName: String?,
-        val isUniqueDisplayName: Boolean,
+        val isUniqueDisplayName: Boolean = false,
```
[the remark this answers is not in the document]
Comment (ganfra) [reply matrix-sdk-android/src/main/java/im/vector/matrix/android/api/session/room/timeline/TimelineEvent.kt]: Yep sorry, it's a miss from some internal testing

**Pull request merged**